src/adapter/src/webhook.rs

```diff
@@ -217,21 +218,110 @@ impl AppendWebhookValidator {
     }
 }
 
+#[derive(Derivative, Clone)]
+#[derivative(Debug)]
 pub struct AppendWebhookResponse {
-    pub tx: MonotonicAppender,
+    /// Channel to monotonically append rows to a webhook source.
+    pub tx: WebhookAppender,
+    /// Column type for the `body` column.
     pub body_ty: ColumnType,
+    /// Types of the columns for the headers of a request.
     pub header_tys: WebhookHeaders,
+    /// Expression used to validate a webhook request.
+    #[derivative(Debug = "ignore")]
     pub validator: Option<AppendWebhookValidator>,
 }
 
-impl fmt::Debug for AppendWebhookResponse {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("AppendWebhookResponse")
-            .field("tx", &self.tx)
-            .field("body_ty", &self.body_ty)
-            .field("header_tys", &self.header_tys)
-            .field("validate_expr", &"(...)")
-            .finish()
+/// A wrapper around [`MonotonicAppender`] that can get closed by the `Coordinator` if the webhook
+/// gets modified.
+#[derive(Clone, Debug)]
+pub struct WebhookAppender {
+    tx: MonotonicAppender,
+    guard: WebhookAppenderGuard,
+}
+
+impl WebhookAppender {
+    /// Checks if the [`WebhookAppender`] has closed.
+    pub fn is_closed(&self) -> bool {
+        self.guard.is_closed()
+    }
+
+    /// Appends updates to the linked webhook source.
+    pub async fn append(&self, updates: Vec<(Row, Diff)>) -> Result<(), AppendWebhookError> {
+        if self.is_closed() {
+            return Err(AppendWebhookError::TemporaryError);
```

Comment:
Uncovered

Reply:
This can only get exercised with a very specific race:

1. environmentd receives webhook request to append, we get a channel from our cache
2. previously running `ALTER SOURCE ... RENAME ...` completes
3. push an update down the channel from our cache

Realistically the amount of time between steps 1 and 3 is about 1ms at most, so testing this is quite difficult.

Reply:
`is_closed()` here checks the guard on the `WebhookAppender`, not the underlying `MonotonicAppender`. The underlying channel to the `storage-controller` remains open, but the Coordinator closes these outstanding `WebhookAppender`s when a modification occurs.